### PR TITLE
Created LGBM classifier @ lgbm_classifier.py

### PR DESCRIPTION
**Pull request by ArchismanKarmakar**
Adding LGBM Classifier Script for this repository.

### Describe your change:

I am adding an implementation of the **LGBM Classifier & LGBM Regressor** using the Iris dataset as an example. This script demonstrates how to use LightGBM to train a model and visualize its performance through a confusion matrix. It includes proper type hints, docstrings, and doctests to ensure clarity and functionality.

* [x] Add an algorithm: LightGBM (LGBMClassifier) and LGBM Regressor example with some example dataset.
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by ArchismanKarmakar** [comment]
@darkstar 
Please check this PR.
lightgbm is not installed.

**Comment by cclauss** [comment]
What does LGBM stand for and why is the reader forced to guess?

**Comment by ArchismanKarmakar** [comment]
> What does LGBM stand for and why is the reader forced to guess?

LGBM stands for Light Gradient Boosting Machine. It's an implementation of gradient boosting that is designed to be highly efficient, fast, and scalable. The name "Light" comes from the fact that it consumes less memory and is faster than traditional gradient boosting implementations like XGBoost. LGBM achieves this by using techniques like histogram-based learning and leaf-wise growth strategies.

No guessing sir, I should have written like this.
# LGBM (Light Gradient Boosting Machine) Classifier Example using Bank Marketing Dataset

@cclauss

**Comment by ArchismanKarmakar** [comment]
@cclauss @cclauss Sir, kindly check this.

**Comment by cclauss** [comment]
Closing tests_are_failing PRs to prepare for Hacktoberfest
